dandi/download.py

```diff
@@ -743,7 +744,7 @@ def _download_file(
     resuming = False
     attempt = 0
     attempts_allowed: int = (
-        3  # number to do, could be incremented if we downloaded a little
+        10  # number to do, could be incremented if we downloaded a little
     )
     while attempt <= attempts_allowed:
         attempt += 1
```

Comment:
`attempt` seems to be used as the index of the attempt, per its used as the argument to `_check_attempts_and_sleep()`, so incrementing it in the beginning of the loop can be problematic. It should be incremented at the end of the loop.

Reply:
I think you can put it in the `finally` clause of the try-statement.

Reply:
thanks for the analysis, but it feels unrelated to the purpose of this PR, or am I missing something?

if you see how to improve overall logic -- please submit a separate PR.  Here the goal is just to increment number of attempts

Reply:
actually, while at it -- I just did /pushed it -- it was a good advice.  But again, next time please just send a separate PR or when "sure" -- you are even welcome to push a commit to my PR to improve it directly!  You have my permission and blessing. If anything -- could be dropped later.